modules/internal/ChapelIteratorSupport.chpl

```diff
@@ -200,29 +219,37 @@ module ChapelIteratorSupport {
       return myvoid;
     }
   }
-  pragma "no doc"
   inline proc chpl_computeIteratorShape(arg) {
     // none of the above cases
     return _void;
   }
-  pragma "no doc"
-  inline proc chpl_iteratorHasShape(ir: _iteratorRecord) param {
+
+  proc chpl_iteratorHasShape(ir: _iteratorRecord) param {
     use Reflection;
     if hasField(ir.type, "_shape_") then
       return ir._shape_.type != void;
     else
       return false;
   }
-  pragma "no doc"
-  inline proc chpl_iteratorFromForExpr(ir: _iteratorRecord) param {
+
+  // This is the static type of chpl_computeIteratorShape(ir).
+  proc chpl_iteratorShapeStaticTypeOrVoid(type ir: _iteratorRecord) type
+  {
+    use Reflection;
+    if hasField(ir, "_shape_") then
+      return __primitive("static field type", ir, "_shape_");
```

Comment:
Could this be PRIM_STATIC_TYPE followed by a PRIM_GET_MEMBER_VALUE ?

Reply:
Gives me "illegal return of value where type is expected".